core/src/main/scala/scalax/collection/GraphTraversalImpl.scala

```diff
@@ -122,10 +121,10 @@ trait GraphTraversalImpl[N, E <: Edge[N]] extends GraphTraversal[N, E] with Trav
     )(_ => edges.slice(0, edges.size - 1))
 
     def result(): Walk = new Walk {
-      val nodes     = self.nodes
-      def edges     = resultEdges
-      val startNode = start
-      val endNode   = nodes(nodes.size - 1)
+      val nodes: mutable.Seq[NodeT] = self.nodes
```

Comment:
The compiler message says "...inferred Iterable[...] instead of ...ArrayBuffer[...]" but this is not an advice to change the type. Actually, here and below, we want to keep the types 2.13 used to infer. This is ok since it's an implementation of the public interface with more generic types.

Reply:
Reverted, the compiler emits +3 error messages

Reply:
What other error messages are you experiencing?
I'd prefer `IndexedSeq` like
```
    def result(): Walk = new Walk {
      val nodes: IndexedSeq[NodeT] = self.nodes
      def edges: IndexedSeq[EdgeT] = resultEdges
      val startNode: NodeT         = start
      val endNode: NodeT           = nodes(nodes.size - 1)
    }
```

Reply:
Changed to `IndexedSeq[...]` and the +3 error messages are solved.

Now, when `scalacOptions ++= Seq("-Xsource:3")` is enabled, there still are these 10 errors:

```
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/AnyGraph.scala:300:57: type mismatch;
[error]  found   : [N](node_1: N, node_2: N, moreNodes: N*): scalax.collection.hyperedges.ordered.HyperEdge[N] <and> [N](ends: scalax.collection.Several[N]): scalax.collection.hyperedges.ordered.HyperEdge[N]
[error]  required: scalax.collection.Several[A] => scalax.collection.generic.AnyHyperEdge[A]
[error]           case _: OrderedEndpoints => ordered.HyperEdge.apply
[error]                                                         ^
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/GraphTraversal.scala:108:14: constructor modifiers are assumed by synthetic `copy` method
[error] Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=scalax.collection.GraphTraversal.Layer
[error]   case class Layer protected[collection] (index: Int, _nodes: IndexedSeq[NodeT]) {
[error]              ^
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/GraphTraversal.scala:210:14: constructor modifiers are assumed by synthetic `copy` method
[error] Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=scalax.collection.GraphTraversal.TopologicalSortFailure
[error]   case class TopologicalSortFailure private[collection] (candidateCycleNodes: Set[NodeT]) {
[error]              ^
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/GraphTraversal.scala:1336:18: constructor modifiers are assumed by synthetic `copy` method
[error] Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=scalax.collection.GraphTraversal.Informer.DfsInformer.Element
[error]       case class Element protected[collection] (node: NodeT, depth: Depth, cumWeight: Double = 0) extends NodeElement
[error]                  ^
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/GraphTraversal.scala:1360:18: constructor modifiers are assumed by synthetic `copy` method
[error] Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=scalax.collection.GraphTraversal.Informer.WgbInformer.Element
[error]       case class Element protected[collection] (
[error]                  ^
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/GraphTraversal.scala:1394:18: constructor modifiers are assumed by synthetic `copy` method
[error] Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=scalax.collection.GraphTraversal.Informer.DijkstraInformer.Element
[error]       case class Element[T: Numeric] protected[collection] (node: NodeT, cumWeight: T, depth: Depth) extends NodeElement
[error]                  ^
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/GraphTraversal.scala:1409:18: constructor modifiers are assumed by synthetic `copy` method
[error] Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=scalax.collection.GraphTraversal.Informer.TarjanInformer.Element
[error]       case class Element protected[collection] (
[error]                  ^
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/hyperedges/ordered/HyperEdge.scala:11:18: constructor modifiers are assumed by synthetic `apply` method
[error] Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=scalax.collection.hyperedges.ordered.HyperEdge
[error] final case class HyperEdge[+N] private (override val ends: Several[N])
[error]                  ^
[warn] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/hyperedges/ordered/HyperEdge.scala:11:54: private default argument in class HyperEdge is never used
[warn] final case class HyperEdge[+N] private (override val ends: Several[N])
[warn]                                                      ^
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/mutable/ArraySet.scala:150:14: constructor modifiers are assumed by synthetic `copy` method
[error] Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=scalax.collection.mutable.ArraySet.CheckedHints
[error]   case class CheckedHints private[ArraySet] (
[error]              ^
[error] /home/mario/IdeaProjects/scala-graph/core/src/main/scala/scalax/collection/mutable/SimpleArraySet.scala:285:16: under -Xsource:3, inferred scala.collection.mutable.Builder[A,scalax.collection.mutable.SimpleArraySet[A]] instead of scala.collection.mutable.GrowableBuilder[A,scalax.collection.mutable.SimpleArraySet[A]] [quickfixable]
[error] Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=scalax.collection.mutable.SimpleArraySet.newBuilder
[error]   override def newBuilder[A]                    = new GrowableBuilder[A, SimpleArraySet[A]](empty)
[error]                ^
[warn] one warning found
[error] 10 errors found
```

Reply:
Down to 9, solved the last one with https://github.com/scala-graph/scala-graph/pull/315/commits/eb8a7ac3648128d37cde0ad8d79b04173cdd1444